cmd/fluent-bit/dque.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/grafana/loki/pkg/promtail/client"
+	"github.com/joncrlsn/dque"
+	"github.com/prometheus/common/model"
+)
+
+type dqueConfig struct {
+	queueDir         string
+	queueSegmentSize int
+	queueSync        bool
+	queueName        string
+}
+
+var defaultDqueConfig = dqueConfig{
+	queueDir:         "/tmp/flb-storage/loki",
+	queueSegmentSize: 500,
+	queueSync:        false,
+	queueName:        "dque",
+}
+
+type dqueEntry struct {
+	Lbs  model.LabelSet
+	Ts   time.Time
+	Line string
+}
+
+func dqueEntryBuider() interface{} {
+	return &dqueEntry{}
+}
+
+type dqueClient struct {
+	logger log.Logger
+	queue  *dque.DQue
+	loki   client.Client
+	quit   chan struct{}
+	once   sync.Once
+	wg     sync.WaitGroup
+}
+
+// New makes a new dque loki client
+func newDque(cfg *config, logger log.Logger) (client.Client, error) {
+	var err error
+
+	q := &dqueClient{
+		logger: log.With(logger, "component", "queue", "host", cfg.clientConfig.TenantID),
```

Comment:
not sure why tenant is named host ? may be a typo ?

Reply:
yes typo it should be queue name, fixed